### ISSUE TITLE
Dutycycle symmetry

### ISSUE THREAD

**Issue by justatry2** [PrincetonUniversity/magnet#100]
Testing 3C94 material with triangular waves under MagNet AI showed that losses are not symmetric in duty cycle. Why?

**Comment by minjiechen** [comment]
The model is fully data driven and we try to avoid putting prior-human knowledge in developing the neural network model. Asymmetries exist in our experimental data, excitation waveform, measurement accuracy, for different duty ratios. As a result, we cannot guarantee full symmetry in the training data, and the model picks up the asymmetry. 

One can develop neural network models that will produce "pure" symmetry data by reasonable hypothesis, but that is beyond our purpose. A data driven model cannot ensure full "symmetry" in experiments and in training, and the material may have intrinsic "asymmetry" due to memory effect (existing theory cannot completely rule out this possibility).

**Comment by justatry2** [comment]
Thanks for your reply.
To my knowledge symmetry of power losses over duty cycle when excited with triangular wave forms is a quite often and consistently reported behavior for ferrites in literature.
So, I am inclined to believe that the model should be modified or trained in the way that it recovers this feature.
However, I am admittedly not an expert in this special field of material characterization and physics.

Von: MInjie Chen ***@***.***>
Gesendet: Montag, 30. Januar 2023 07:12
An: PrincetonUniversity/magnet ***@***.***>
Cc: Kapoor Jivan (AE/ENS2) ***@***.***>; Author ***@***.***>
Betreff: Re: [PrincetonUniversity/magnet] Dutycycle symmetry (Issue #100)


The model is fully data driven and we try to avoid putting prior-human knowledge in developing the neural network model. Asymmetries exist in our experimental data, excitation waveform, measurement accuracy, for different duty ratios. As a result, we cannot guarantee full symmetry in the training data, and the model picks up the asymmetry.

One can develop neural network models that will produce "pure" symmetry data by reasonable hypothesis, but that is beyond our purpose. A data driven model cannot ensure full "symmetry" in experiments and in training, and the material may have intrinsic "asymmetry" due to memory effect (existing theory cannot completely rule out this possibility).

**Comment by minjiechen** [comment]
Yes agreed! A model that can leverage both the strengths of data driven models and physical understanding would perform the best. We leave it to the community to find the best way to combine them. 

For now, we will keep our model 100% data driven to perhaps highlight / warn the weakness of a pure data driven model - the biases / inaccuracy in data will directly transfer to biases / inaccuracy in models.
